joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/datastructures/CSharpScope.scala

```diff
@@ -119,9 +127,20 @@ class CSharpScope(typeMap: TypeMap) extends Scope[String, DeclarationNew, ScopeT
     */
   override def pushNewScope(scopeNode: ScopeType): Unit = {
     scopeNode match
-      case NamespaceScope(fullName) => typesInScope.addAll(typeMap.classesIn(fullName))
-      case TypeScope(name, _)       => typesInScope.addAll(typeMap.findType(name))
-      case _                        =>
+      case NamespaceScope(fullName) => {
```

Comment:
Could we rather move this logic under `astForNamespaceDeclaration` or something similar? Also, a potentially cleaner solution that avoids blank strings (that could be placed under `astForNamespaceDeclaration`) could be something like
```scala
def recurseNamespace(parts: List[String], prefix: List[String] = List.empty): Unit = {
  parts match {
    case head :: tail => 
      val currentFullName = prefix :+ head
      scope.pushNewScope(NamespaceScope(currentFullName.mkString(".")))
      recurseNamespace(tail, currentFullName)
    case Nil => // nothing
  }
}
val namespaceParts = fullName.split("[.]").toList
recurseNamespace(namespaceParts)
// Do ast stuff
namespaceParts.foreach(_ => scope.popScope())
```
Note the popping at the end is also important.